### ISSUE TITLE
Allow flexible location of lv_conf.h

### ISSUE THREAD

**Issue by hydra** [lvgl/lvgl#294]
Requiring a fixed location for lv_conf.h based on a relative path isn't flexible enough to allow the user to choose the location of the lvgl source tree in their application.

it's common to find projects with a `lib` directory which contains required libraries, this `lib` directory is often located as a sibling to `src` directory.

developers, by convention, then know that `lib` is essentially read-only, and `src` will contain the user's own code.  library updates modify `lib`, application changes modify `src`.

**Comment by hydra** [comment]
I pushed a commit which was all I required to enable lvgl to be compiled from a `lib` folder while the `lv_conf.h` file was moved inside the `src` folder. The include path in the Makefile was simply updated so the compiler finds the `lv_conf.h` when building the lvgl sources.

https://github.com/hydra/lvgl/commit/4073e1473e98bc499ad3dff4fd943c38cdfbd95f

Likely all the demo projects, etc, will need updated to cope with such a change.
